.github/workflows/docker_publish.yml

```diff
@@ -57,17 +57,36 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Configure docker image tag and build arguments
+        run: |
+          image_base_tag=ghcr.io/${{ github.repository_owner }}/dagmc-ci-ubuntu-${{ matrix.ubuntu_version }}-${{ matrix.compiler}}-hdf5_${{ matrix.hdf5_version}}-moab_${{ matrix.moab_version }}
+          if [ ${{ matrix.geant4_version }} != "OFF" ] ; then
+            image_base_tag=${image_base_tag}-geant4_${{ matrix.geant4_version }}
+            GEANT4=ON
+          else
+            GEANT4=OFF
+          fi
+          if [ ${{ matrix.double_down_version }} != "OFF" ] ; then
+            image_base_tag=${image_base_tag}-double_down_${{ matrix.double_down_version }}
+            DOUBLE_DOWN=ON
+          else
+            DOUBLE_DOWN=OFF
+          fi
```

Comment:
See above comments about always tagging with `OFF` or the version number

Reply:
This block checks if the `matrix.geant4_version` is not OFF. If true, it appends the Geant4 version to the `image_base_tag` and sets the GEANT4 variable to ON; otherwise, it sets GEANT4 to OFF. This variable will let install Geant4 in the docker image and install DAGMC with Geant4 support later. Same for Double Down.

Reply:
I understand what it's doing, but I don't think it's necessary.  We can overload the purpose of the `geant4_version` variable and assign it a value of `OFF` or a value of the version in question.  We can then check that in the Dockerfile to either skip building it or building the requested version.  It creates fewer opportunities for these two variables to break inconsistent.